src/methods/handleIncoming.ts

```diff
@@ -7,23 +7,33 @@ const handleIncoming = (subscriptions: Subscriptions, ...action: [type: Incoming
 
   const [type, payload] = action;
 
-  if( type === "error"){
+  if (type === "error") {
     console.error(payload);
     return;
   }
 
+  // Track summary payload also has the details about visible transcripts
+  // which we will be treating as a separate incoming action of its own.
+  if (type === IncomingActionType.TRACK_SUMMARY) {
+    handleIncoming(
+      subscriptions,
+      IncomingActionType.VISIBLE_TRANSCRIPTS,
+      payload
+    )
+  }
+
   const subscriptionsToAction = subscriptions.get(type);
 
   if (subscriptionsToAction) {
-      
-    [...subscriptionsToAction.values()].forEach( subscription => {
-    
-    const formattedIncoming = formatIncoming(type, payload);
-    if(!formattedIncoming){
-      console.error(payload);
-      return;
-    }
-    subscription(formattedIncoming);
+
+    [...subscriptionsToAction.values()].forEach(subscription => {
+
+      const formattedIncoming = formatIncoming(type, payload);
+      if (!formattedIncoming) {
+        console.error(payload);
+        return;
+      }
```

Comment:
out of curiosity, why do you repeatedly format the incoming message inside the forEach loop? What if you formatted it once outside the loop? And didn't call the subscribed functions if there is no formattedIncoming?